### PR TITLE
Add Firebase deploy recipe (Fixes #508)

### LINKED ISSUE #508: Add Firebase Deploy Recipe for Pretty URLs
Add Firebase Deploy Recipe for Pretty URLs to /docs.

### PR DESCRIPTION
**Pull request by chuckh**
This has borrowed heavily from work that @kaycebasques has done for
Polymer Docs.  I just added the pretty URLs stuff. Fixes #508

### CODE REVIEW

File: docs/deploy-to-firebase-pretty-urls.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          ],
+          "rewrites": [ {
+            "source": "**",
+            "destination": "/index.html"
+          } ]
+        }
+
+1.  Add `<base href="/">` to `head` in index.html
+
+1.  Set `hashbang: false` in routing.html near bottom
```
Comment (robdodson) [inline docs/deploy-to-firebase-pretty-urls.md]: I think you could just remove the hashbang option all together and just call `page();`

**Comment by mbleigh** [comment]
Started related discussion in #512 -- happy to help with any Firebase related questions!

**Comment by addyosmani** [comment]
Thanks for opening that issue, @mbleigh! We're going to target this recipe for the release right after the next one so there's a little time to figure out a good story there.

**Comment by addyosmani** [comment]
Friendly reminder that this requires a rebase against master once we've resolved the discussion in https://github.com/PolymerElements/polymer-starter-kit/issues/512

**Comment by chuckh** [comment]
It has been rebased. We might want to go ahead and merge this recipe since it works and developers are using/referencing it on Slack.

We can save moving bower_components and using Superstatic middleware for later pull requests.

**Comment by samccone** [comment]
:+1: to landing :airplane:

**Comment by addyosmani** [comment]
LGTM

**Pull request merged**